File: templates/run_IBD_QC_qcplink.pl
#!/usr/bin/env perl

use strict;

my %imiss;
my %removed;

open IMISS, '<', "$missing"
    or die "Cannot open missing file ($missing): \$!\n";

print "Reading PLINK .imiss file $missing\n";

while(<IMISS>){
    s/^\\s+//;
    my @fields = split /\\s+/, \$_;
    \$imiss{\$fields[0]}{\$fields[1]} = \$fields[5];
}

open GENOME, "<$ibd_genome"
    or die "Cannot open genotypes file ($ibd_genome): \$!\n";
print "Reading PLINK .genome file $ibd_genome\n";
open OUT, ">$outfname";
while(<GENOME>){
    s/^\\s+//;
    my @fields = split /\\s+/, \$_;
    if(\$fields[9] > 0.185){
	if(\$imiss{\$fields[0]}{\$fields[1]}>\$imiss{\$fields[2]}{\$fields[3]}){
	    unless(\$removed{\$fields[0]}{\$fields[1]}){
		print OUT "\$fields[0] \$fields[1]\n";
		\$removed{\$fields[0]}{\$fields[1]} = 1;
	    }
	}
	elsif(\$imiss{\$fields[0]}{\$fields[1]}<\$imiss{\$fields[2]}{\$fields[3]}){
	    unless(\$removed{\$fields[2]}{\$fields[3]}){
		print OUT "\$fields[2] \$fields[3]\n";
		\$removed{\$fields[2]}{\$fields[3]} = 1;
	    }
	}
	else{
	    unless(\$removed{\$fields[0]}{\$fields[1]}){
		print OUT "\$fields[0] \$fields[1]\n";
		\$removed{\$fields[0]}{\$fields[1]} = 1;
	    }
	}
    }
}

File: templates/run_IBD_QC_qcplink.py
#!/usr/bin/env python3

imissf  = open("$missing")

EOL=chr(10)
imiss={}
removed={}
imissf.readline()
for line in imissf:
    fields = line.strip().split()
    imiss[fields[0]+" "+fields[1]] = float(fields[5]);

genomef=open("$ibd_genome")
outf   =open("$outfname","w")
genomef.readline()
for line in genomef:
    fields = line.strip().split()
    id1 = fields[0]+" "+fields[1]
    id2 = fields[2]+" "+fields[3]
    if(float(fields[9]) > ${pi_hat}):
	if imiss[id1] >= imiss[id2] and id1 not in removed:
           outf.write("%s%s"%(id1,EOL))
           removed[id1]=1;
	elif imiss[id1]<imiss[id2] and id2 not in removed:
           outf.write("%s%s"%(id2,EOL));
           removed[id2]=1;
outf.close()
